src/version/version_test.go

```diff
@@ -0,0 +1,93 @@
+//go:build unit
+// +build unit
+
+package version_test
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/NeowayLabs/semantic-release/src/log"
+	"github.com/NeowayLabs/semantic-release/src/tests"
+	"github.com/NeowayLabs/semantic-release/src/version"
+)
+
+type fixture struct {
+	versionControl *version.VersionControl
+}
+
+func setup() *fixture {
+	logger, err := log.New("test", "", "info")
+	if err != nil {
+		errors.New("error while getting new log")
+	}
+
+	return &fixture{versionControl: version.NewVersionControl(logger, PrintElapsedTimeMock)}
+}
+
+func PrintElapsedTimeMock(what string) func() {
+	return func() {
+		fmt.Println("print elapsed time")
+	}
+}
+
+func TestGetNewVersionGetCommitChangeTypeFromMessageError(t *testing.T) {
+	f := setup()
+	actualVersion, actualErr := f.versionControl.GetNewVersion("", "")
+	tests.AssertError(t, actualErr)
+	tests.AssertEqualValues(t, "error while finding commit change type within commit message due to: change type not found", actualErr.Error())
+	tests.AssertEmpty(t, actualVersion)
+}
+
+func TestGetNewVersionSplitVersionMajorMinorPatchError(t *testing.T) {
+	f := setup()
+	actualVersion, actualErr := f.versionControl.GetNewVersion("type:[feat]", "1.0.a")
+	tests.AssertError(t, actualErr)
+	tests.AssertEqualValues(t, "error while spliting version into MAJOR.MINOR.PATCH due to: could not convert a to int", actualErr.Error())
+	tests.AssertEmpty(t, actualVersion)
+}
+
+func TestGetNewVersionGetUpgradeTypeError(t *testing.T) {
+	f := setup()
+	actualVersion, actualErr := f.versionControl.GetNewVersion("type:[skip]", "1.0.0")
+	tests.AssertError(t, actualErr)
+	tests.AssertEqualValues(t, "error while getting upgrade type due to: skip is an invalid upgrade change type", actualErr.Error())
+	tests.AssertEmpty(t, actualVersion)
+}
+
+func TestGetNewVersionMajorSuccess(t *testing.T) {
+	f := setup()
+	actualVersion, actualErr := f.versionControl.GetNewVersion("type:[breaking change]", "1.0.0")
+	tests.AssertNoError(t, actualErr)
+	tests.AssertEqualValues(t, "2.0.0", actualVersion)
+}
+
+func TestGetNewVersionMinorSuccess(t *testing.T) {
+	f := setup()
+	actualVersion, actualErr := f.versionControl.GetNewVersion("type:[feat]", "1.0.0")
+	tests.AssertNoError(t, actualErr)
+	tests.AssertEqualValues(t, "1.1.0", actualVersion)
+}
+
+func TestGetNewVersionPatchSuccess(t *testing.T) {
+	f := setup()
+	actualVersion, actualErr := f.versionControl.GetNewVersion("type:[fix]", "1.0.0")
```

Comment:
Is there any chance of having a broken `currentVersion` as an input, like "1.0"? maybe add a test just to make sure it won't panic

Reply:
Yeah...
I've added a new validation and a test covering this part.